ext/page_log/palm/palm.c

```diff
@@ -524,11 +524,11 @@ palm_add_reference(WT_PAGE_LOG *page_log)
     palm = (PALM *)page_log;
 
     /*
-     * Missing reference or overflow?
+     * Missing reference or overflow? Assume the first ref will not race the creation.
      */
     if (palm->reference_count == 0 || palm->reference_count + 1 == 0)
         return (EINVAL);
-    ++palm->reference_count;
+    (void)__wt_atomic_add32(&palm->reference_count, 1);
```

Comment:
In extensions we can use standard C11 atomics (from `<stdatomic.h>`): `atomic_fetch_{add|sub}`